modules/client/src/erc20/schemas.ts

```diff
@@ -0,0 +1,26 @@
+import { array, number, object, string } from "yup";
+import {
+  addressOrEnsSchema,
+  bigintSchema,
+  pluginSettingsSchema,
+} from "../client-common";
+
+export const newTokenParamsSchema = object({
+  name: string().required(),
+  symbol: string().required(),
+  decimals: number().positive().required(),
+  minter: addressOrEnsSchema,
+  balances: array(object({
+    address: addressOrEnsSchema.required(),
+    balance: bigintSchema.required(),
+  })).required(),
+});
+export const existingTokenParamsSchema = object({
+  address: addressOrEnsSchema.required(),
+});
+
+export const erc20PluginInstallSchema = object({
+  settings: pluginSettingsSchema.required(),
+  newToken: newTokenParamsSchema.default(undefined),
+  useToken: existingTokenParamsSchema.default(undefined),
```

Comment:
Is `settings` the common settings and `newtoken` and `useToken` the custom bit?

I suggest we have a 100% custom settings object. SDK users should be building a single settings object, regardless of the code that we can reuse internally. 

This may also include the `earlyExecution` flag and the `voteReplacement` one as well